### PR TITLE
Align

### PR DESCRIPTION
**Pull request by NicoRenaud**
Integrate the align capabilities of pdb2sql in deeprank. Allows to do something like :
```python
database = DataGenerator(
    pdb_source=pdb_source,
    pdb_native=pdb_native,
    pssm_source=pssm_source,
    align={"axis":'z','export':False},
    data_augmentation=2,
    compute_targets=[
        'deeprank.targets.dockQ',
        'deeprank.targets.binary_class'],
    compute_features=[
        'deeprank.features.AtomicFeature',
        'deeprank.features.FullPSSM',
        'deeprank.features.PSSM_IC',
        'deeprank.features.BSA',
        'deeprank.features.ResidueDensity'],
    hdf5=h5file,
    mpi_comm=comm)
```
there all the complexes will be aligned in the z direction.  All data augmentation is done in the plane prepandicular to the alignement axis.

**NOTE** : you need the `align` branch of `pdb2sql` as the align features is not yet merged in the the `master` of `pdb2sql`

### CODE REVIEW

File: deeprank/generate/DataGenerator.py [hunk 1]
```diff
@@ -33,7 +35,7 @@ def _printif(string, cond): return print(string) if cond else None
 class DataGenerator(object):
 
     def __init__(self, pdb_select=None, pdb_source=None,
-                 pdb_native=None, pssm_source=None,
+                 pdb_native=None, pssm_source=None, align=None,
```
Comment (CunliangGeng) [inline deeprank/generate/DataGenerator.py]: It lacks docstring for `align`.

File: deeprank/generate/DataGenerator.py [hunk 2]
```diff
@@ -239,6 +242,7 @@ def create_database(
 
             # names of the molecule
             mol_name = os.path.splitext(os.path.basename(cplx))[0]
+            mol_name = mol_name.replace('-', '_')
```
Comment (CunliangGeng) [inline deeprank/generate/DataGenerator.py]: Why replacing `-` with `_` for molecule name?

File: test/test_generate.py [hunk 3]
```diff
@@ -183,9 +183,72 @@ def test_4_add_feature(self):
...
+            database.create_database(prog_bar=True, random_seed=2019)
+            print(' ' * 25 + '--> Done in %f s.' % (time() - t0))
+        else:
+            print('{:25s}'.format('Use existing database') + database.hdf5)
+
+
 if __name__ == "__main__":
-    # unittest.main()
 
+    # unittest.main()
```
Comment (CunliangGeng) [inline test/test_generate.py]: Could you add the new test cases here? I run it locally, tests passed.

File: deeprank/generate/DataGenerator.py (hunk 2 above)
Comment (NicoRenaud) [reply deeprank/generate/DataGenerator.py]: the dash causes issues in DeepXplorer and the simplest issue was to replace them with underscore

File: deeprank/generate/DataGenerator.py (hunk 1 above)
Comment (NicoRenaud) [reply deeprank/generate/DataGenerator.py]: done

File: test/test_generate.py (hunk 3 above)
Comment (NicoRenaud) [reply test/test_generate.py]: done

File: deeprank/generate/DataGenerator.py (hunk 2 above)
Comment (CunliangGeng) [reply deeprank/generate/DataGenerator.py]: I see, thanks

File: deeprank/generate/DataGenerator.py (hunk 1 above)
Comment (CunliangGeng) [reply deeprank/generate/DataGenerator.py]: add one more example for interface alignment?

**Pull request merged**